types/unist/index.d.ts

```diff
@@ -79,19 +78,21 @@ export interface Point {
     offset?: number | undefined;
 }
 
+export type NodeData<T extends Node> = T extends Node<infer R> ? R : never;
+
 /**
  * Nodes containing other nodes.
  */
-export interface Parent extends Node {
+export interface Parent<T extends Node = Node, Data = NodeData<T>> extends Node<Data> {
```

Comment:
Could we add a test case with nested parents with different types of data?
I'm hopeful this can handle the recursive case, but a test case would help ensure we don't cause breakage.

Reply:
See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54413/files#r667140524

Reply:
That isn't nested parents with different types.
Which if I'm reading your example correctly would look something like

```ts
const twoParentsAndALiteral: Parent<Parent<Literal<string>, typeof data>, typeof anotherData>
```

Reply:
OK

Reply:
See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54413/files#diff-e15dc003e82168ec12b1ff0d31c0afe72a42b65a74d8aa9d9ae28bbce378fafdR110